### PR TITLE
Fix schedule manager to adjust child activities according to parent changes

### PR DESCRIPTION
**Pull request by jonatanklosko**
Fixes https://github.com/jonatanklosko/groupifier-next/issues/1 (the issue was submitted on that repo, but it's a bug on the WCA side).

The implementation moves child activities when the parent one is moved, and scales them by the same rate if the parent is resized. I think that's the most reasonable approach here.

Additionally I fixed extensions not to be wiped out when they are missing in an update request body.

### CODE REVIEW

File: WcaOnRails/app/javascript/edit-schedule/SchedulesEditor/calendar-utils.js [hunk 1]
```diff
@@ -96,7 +110,7 @@ function ambiguousMomentToIsoString(editor, momentObject) {
   let tz = venue.timezone;
   // Take the moment and "concatenate" the UTC offset of the timezone at that time
   // momentObject is a FC (ambiguously zoned) moment, therefore format() returns a zone free string
-  return moment.tz(momentObject.format(), tz).format();
+  return moment.tz(momentObject.format(), tz).utc().format();
```
Comment (viroulep) [inline WcaOnRails/app/javascript/edit-schedule/SchedulesEditor/calendar-utils.js]: Why is that necessary?

Comment (jonatanklosko) [reply WcaOnRails/app/javascript/edit-schedule/SchedulesEditor/calendar-utils.js]: It's not necessary, but I did it for the sake of consistency. Previously it would return an activity with `startDate` and `endDate` as ISO strings with timezone, whereas initially it's always UTC.

Comment (viroulep) [reply WcaOnRails/app/javascript/edit-schedule/SchedulesEditor/calendar-utils.js]: I wish I'd remember why I did it like that, but it seems good to me now.
